app/src/main/java/com/android/sample/ui/profile/Profile.kt

```diff
@@ -222,25 +238,34 @@ fun ProfileContent(
 fun LazyListScope.displayActivitySection(
     sectionTitle: String,
     category: String,
-    user: User,
-    listActivitiesViewModel: ListActivitiesViewModel,
+    listActivities: List<Activity>,
     navigationActions: NavigationActions,
-    userProfileViewModel: ProfileViewModel
+    userProfileViewModel: ProfileViewModel,
+    listActivitiesViewModel: ListActivitiesViewModel,
+    isParticipantProfile: Boolean,
+    user: User
 ) {
   item {
     Spacer(modifier = Modifier.height(MEDIUM_PADDING.dp))
     SectionTitle(title = sectionTitle, testTag = "${category}ActivitiesTitle")
   }
-
-  user.activities?.let { activities ->
-    items(activities.size) { index ->
+  if (listActivities.isNotEmpty()) {
+    items(listActivities.size) { index ->
       ActivityBox(
-          activityId = activities[index],
-          user = user,
+          activity = listActivities[index],
           listActivitiesViewModel = listActivitiesViewModel,
           navigationActions = navigationActions,
           category = category,
-          userProfileViewModel = userProfileViewModel)
+          userProfileViewModel = userProfileViewModel,
+          user = user)
+    }
+  } else {
+    if (!isParticipantProfile) {
+      item { PlusButtonToCreate(navigationActions = navigationActions) }
```

Comment:
How about if the enrolled activities is empty, we will still have a plus button that navigates to create activity. This makes no sens. I suggest you to use the category argument to distinguish between the 3 sections.